Automattic/jetpack
# Fixes missing controller for DSP /subscriptions POST endpoint

This PR fixes a problem in the Blaze Dashboard when a user tries to create a campaign and store their payment method.
The root of the problem is a missing controller in the Blaze package for the `/subscriptions` endpoint.

## Proposed changes:
* Adds the missing controller mapping in the Blaze package

### Other information:


## Jetpack product discussion


N/A

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






The Blaze Dashboard can be tested on self-hosted and pressable sites. You can also test this on atomic, but you will need to manually execute a filter: add_filter( 'jetpack_blaze_dashboard_enable', '__return_true' );.

* Go to Tools->Advertising
* Open your Network console tab and filter requests with the `/subscriptions` path
* Click on the Promote button in any of your posts
* If you got to the Express flow, you will need to click the "**Make Changes**" button
* In the Full flow, fill all the required information
* In the payment section, you need to input a valid credit card and then check the "**Save this card for future payments**" checkbox. (If you don't see the credit card section, you can click the "**Add new card**" link).
* Click on the "Submit Campaign" button
* Verify that the call to `/subscriptions` ended correctly, and that the campaign was successfully created.
* You can go to the campaign details and cancel the campaign if you want
